constellations/gemini-admin/src/session/sessionAdd.js

```diff
@@ -0,0 +1,175 @@
+"use strict";
+require("./sessionAdd.sass");
+import React from "react";
+import moment from "moment";
+
+// React DatePicker
+import "react-dates/initialize";
+import "react-dates/lib/css/_datepicker.css";
+import { SingleDatePicker } from "react-dates";
+
+// React TimePicker
+import "react-times/css/classic/default.css";
+import TimePicker from "react-times";
+
+export class SessionAdd extends React.Component {
+    constructor(props) {
+        super(props);
+        // Default is 3:00 pm
+        let today = moment().hour(15).minute(0);
+        this.state = {
+            dateFocused: false,
+            numWeeks: 1,
+            startsAt: today,
+            endsAt: moment(today).add(2, "h"),
+        };
+    }
+
+    onDateChange = (date) => {
+        let newDate = moment(this.state.startsAt)
+            .date(date.date())
+            .month(date.month())
+            .year(date.year());
+
+        this.setState({
+            startsAt: newDate,
+        });
+    };
+
+    onTimeChange = (inputField, options) => {
+        let { hour, minute, meridiem } = options;
+        hour = parseInt(hour);
+        minute = parseInt(minute);
+        let newHour =
+            meridiem == "PM" && parseInt(hour) < 12 ? hour + 12 : hour;
+
+        if (inputField == "start") {
+            const newStartTime = moment(this.state.startsAt)
+                .hour(newHour)
+                .minute(minute)
+                .second(0);
+            const newEndTime = moment(newStartTime).add(2, "h");
+
+            this.setState({
+                startsAt: newStartTime,
+                endsAt: newEndTime,
+            });
+        } else if (inputField == "end") {
+            const newEndTime = moment(this.state.endsAt)
+                .hour(newHour)
+                .minute(minute)
+                .second(0);
+
+            this.setState({
+                endsAt: newEndTime,
+            });
+        }
+    };
+
+    onRepeatChange = (e) => {
+        this.setState({
+            numWeeks: parseInt(e.target.value == "" ? 0 : e.target.value),
+        });
+    };
+
+    onClickAddSessions = () => {
+        var sessions = [];
+        for (let i = 0; i < this.state.numWeeks; i++) {
+            let startsAt = moment(this.state.startsAt).add(i, "w");
+            let endsAt = moment(this.state.endsAt).add(i, "w");
+
+            sessions.push({
+                startsAt: startsAt,
+                endsAt: endsAt,
+            });
+        }
+        this.props.addSessions(sessions);
+    };
+
+    render = () => {
+        return (
+            <section id="add-sessions">
+                <h3>Add Sessions</h3>
+
+                <div className="container">
+                    <div id="date-inputs">
+                        <div className="row">
+                            <div className="item">
```

Comment:
Is there any way to use less divs here?

Reply:
Not really. The item div is necessary to keep the label and the date picker together. The row is a flexbox. The `date-inputs` holds the 4 input things together and the `container` is another flexbox that helps with formatting the button next to the inputs.